folium/folium.py

```diff
@@ -664,6 +666,31 @@ def fit_bounds(self, bounds, padding_top_left=None,
         self.template_vars.update({'fit_bounds': fit_bounds_str})
 
 
+    def _auto_bounds(self):
+        if 'fit_bounds' in self.template_vars:
+            return
+        # Get count for each feature type
+        ft_names = ["marker", "line", "circle", "polygon", "multiline"]
+        ft_names = [i for i in ft_names if i in self.mark_cnt]
+
+        # Make a comprehensive list of all the features we want to fit
+        feat_str = ["{name}_{count}".format(name=ft_name,
+                                            count=self.mark_cnt[ft_name])
+                    for ft_name in ft_names
+                    for count in xrange(1, self.mark_cnt[ft_name]+1)
```

Comment:
It is fine to use range here. I do not believe it will ever be a memory problem in python2 and in python3 range is a generator.